### ISSUE TITLE
missing field `discriminator` warnings

### ISSUE THREAD

**Issue by Miezhiko** [serenity-rs/serenity#2022]
branch: `next`

```
WARN run:recv_event:handle_event{event=Err(Json(Error("missing field `discriminator`", line: 0, column: 0)))}: serenity::gateway::shard: [ShardInfo { id: 0, total: 1 }] Unhandled error: Json(Error("missing field `discriminator`", line: 0, column: 0))
```

seems like 
```
{
  "t": "PRESENCE_UPDATE",
  "s": 138,
  "op": 0,
  "d": {
    "user": {
      "id": "..."
    },
    "status": "...",
    "guild_id": "...",
    "client_status": {},
    "activities": [...]
  }
}
```
is the payload minified

possibly related: https://github.com/serenity-rs/serenity/commit/ca712d09540a424464fe77427791d16f53da131e

**Comment by Miezhiko** [comment]
this fixes my problem <https://github.com/Qeenon/serenity/commit/36a0d9e7b683e2e61b208671e42f2a6621618321>

but I may guess that it's not correct way

**Comment by nickelc** [comment]
I see the problem.

> this fixes my problem [Qeenon@36a0d9e](https://github.com/Qeenon/serenity/commit/36a0d9e7b683e2e61b208671e42f2a6621618321)
> 
> but I may guess that it's not correct way

Better workaround is change to it to `#[serde(default, with = "discriminator::option")]`.

I'm looking into a fix for `discriminator::option`.

**Comment by Miezhiko** [comment]
ye I've tried something like that <https://github.com/Qeenon/serenity/commit/72f9dd497bc300f4b279b5304537f2a1a2c5adcc>
but in the end decided to just the cut code for now

maybe will try your fix some later.
